web/html/src/manager/virtualization/pools/list/pools-list.js

```diff
@@ -0,0 +1,191 @@
+                      <i className="fa fa-check-square fa-1-5x" title={t(`${item.data.name} is persistent`)}/>
+                    }
+                  </CustomDiv>,
+                  <CustomDiv key="target" className="col" width="30" um="%">{item.data.targetPath}</CustomDiv>,
+                  <CustomDiv key="usage" className="col" width="10" um="%">
+                    <ProgressBar
+                      progress={Math.round(100 * item.data.allocation / item.data.capacity)}
+                      title={t(`${item.data.allocation / (1024 * 1024)} GiB / ${item.data.capacity / (1024 * 1024)} GiB in use`)}
+                    />
+                  </CustomDiv>,
+                  <CustomDiv key="actions" className="col text-right" width="10" um="%">
+                  </CustomDiv>,
+              ];
+            }
+
+          function renderVolume (item: Tree.TreeItem, renderNameColumn: Function): React.Node {
+            return [
+              <CustomDiv key="name" className="col" width="30" um="%">
+                {renderNameColumn(item.data.name)}
+              </CustomDiv>,
+              <CustomDiv key="usedBy" className="col" width="30" um="%">{item.data.usedBy != null ? item.data.usedBy.join(", ") : ''}</CustomDiv>,
+              <CustomDiv key="usage" className="col" width="10" um="%">
+                <ProgressBar
+                  progress={Math.min(100, Math.round(100 * item.data.allocation / item.data.capacity))}
+                  title={t(`${item.data.allocation / (1024 * 1024)} GiB / ${item.data.capacity / (1024 * 1024)} GiB in use`)}
+                />
+              </CustomDiv>,
+            ];
+          }
+
+          function renderItem (item: Tree.TreeItem, renderNameColumn: Function): React.Node {
+            const renderers = {
+              pool: renderPool,
+              volume: renderVolume,
+            };
+            return renderers[item.data.itemType](item, renderNameColumn);
+          }
+
+          const tree = poolsInfoToTree(pools);
+          const header = [
+            <CustomDiv key="name" className="col" width="30" um="%">{t('Name')}</CustomDiv>,
+            <CustomDiv key="state" className="col text-center" width="5" um="%">{t('State')}</CustomDiv>,
+            <CustomDiv key="autostart" className="col text-center" width="5" um="%">{t('Autostart')}</CustomDiv>,
+            <CustomDiv key="persistent" className="col text-center" width="5" um="%">{t('Persistent')}</CustomDiv>,
+            <CustomDiv key="target" className="col" width="30" um="%">{t('Location')}</CustomDiv>,
+            <CustomDiv key="usage" className="col" width="10" um="%">{t('Usage')}</CustomDiv>,
+            <CustomDiv key="actions" className="col text-right" width="10" um="%">{t('Actions')}</CustomDiv>,
+          ];
+
+          return (
+            <DataHandler
+              data={getPools(tree)}
+              identifier={(raw) => raw.id}
+              initialItemsPerPage={userPrefPageSize}
```

Comment:
argh I don't like this global variable :P can't we just inject it through a prop by the renderer function?

using the same as here: https://github.com/uyuni-project/uyuni/blob/f81adcb9cbf8cc2156c992daa8436ed34be25182/java/code/src/com/suse/manager/webui/templates/common.jade#L10

Reply:
done

Reply:
we could change the `userPrefPageSize` to `props.pageSize` and also delete the line: https://github.com/uyuni-project/uyuni/pull/684/files#diff-aedd239ee9db3dd05408cae2fb2e844dR8